includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -358,7 +358,7 @@ public function getNew(){
 			$ns = array_unique( $ns );
 		}
 
-		$key = self::getCacheKey( self::NEW_ARTICLES_CACHE_ID, '', [ implode( '-', $ns ) , $minArticleQuality ] );
+		$key = self::getCacheKey( self::NEW_ARTICLES_CACHE_ID, '', [ implode( '-', $ns ) , $minArticleQuality, $limit ] );
```

Comment:
How many different values can `$limit` have? Maybe it would be better to set a maximum value of `$limit`, cache the whole results set and slice cached results when requested with smaller limit.